snapd.go

```diff
@@ -249,13 +249,11 @@ func action(ctx *cli.Context) {
 	setMaxProcs(cfg.GoMaxProcs)
 
 	c := control.New(cfg.Control)
-
-	coreModules = []coreModule{}
-
-	coreModules = append(coreModules, c)
 	s := scheduler.New(cfg.Scheduler)
+	c.SetTaskScheduler(s)
 	s.SetMetricManager(c)
-	coreModules = append(coreModules, s)
+
+	coreModules = []coreModule{c, s}
```

Comment:
>So why talking in opposite way c.SetTaskScheduler(s) can't be implemented?

It's not that it can't it's that, at least imo, it shouldn't. Control shouldn't need to know/care/think about tasks. 

>Maybe just push this 2-way communication into already existing s.SetMetricManager(c), so after setting metric manager, scheduler will be able to talk to metric manager (as it works now) and vice versa?

What are you thinking is the way to add 2-way communication to that interface?